packages/react/src/forms/Checkbox/Checkbox.test.tsx

```diff
@@ -103,4 +107,321 @@ describe('Checkbox', () => {
+    expect(span).toHaveClass('Checkbox', 'my-custom-checkbox')
+  })
+
+  it('renders with custom id', () => {
+    const {getByRole} = render(<Checkbox id="my-checkbox-id" value="test" />)
+
+    const checkbox = getByRole('checkbox')
+    expect(checkbox).toHaveAttribute('id', 'my-checkbox-id')
+  })
+
+  it('generates an id when id prop is not provided', () => {
+    const {getByRole} = render(<Checkbox value="test" />)
+
+    const checkbox = getByRole('checkbox')
+    expect(checkbox).toHaveAttribute('id')
+  })
+
+  it('renders with arbitrary attributes', () => {
+    const {getByRole} = render(
+      <Checkbox value="test" data-testid="custom-checkbox" aria-describedby="helper-text" title="My checkbox" />,
+    )
+
+    const checkbox = getByRole('checkbox')
+    expect(checkbox).toHaveAttribute('data-testid', 'custom-checkbox')
+    expect(checkbox).toHaveAttribute('aria-describedby', 'helper-text')
+    expect(checkbox).toHaveAttribute('title', 'My checkbox')
+  })
+
+  it('is unchecked by default', () => {
+    const {getByRole} = render(<Checkbox value="test" />)
+
+    const checkbox = getByRole('checkbox')
+    expect(checkbox).not.toBeChecked()
+    expect(checkbox).not.toHaveAttribute('aria-checked')
+  })
+
+  it('is checked when checked prop is true', () => {
+    const {getByRole} = render(<Checkbox value="test" checked onChange={jest.fn()} />)
+
+    const checkbox = getByRole('checkbox')
+    expect(checkbox).toBeChecked()
+    expect(checkbox).toHaveAttribute('aria-checked', 'true')
+  })
+
+  it('is unchecked when checked prop is false', () => {
+    const {getByRole} = render(<Checkbox value="test" checked={false} onChange={jest.fn()} />)
+
+    const checkbox = getByRole('checkbox')
+    expect(checkbox).not.toBeChecked()
+    expect(checkbox).toHaveAttribute('aria-checked', 'false')
+  })
+
+  it('has non-indeterminate state by default', () => {
+    const {getByRole, container} = render(<Checkbox value="test" />)
+
+    const checkbox = getByRole('checkbox') as HTMLInputElement
+    const span = container.querySelector('span.Checkbox')
+    const checkmarkPath = container.querySelector('path.Checkbox-checkmark-path')
+    const dashPath = container.querySelector('path:not(.Checkbox-checkmark-path)')
+
+    expect(checkbox.indeterminate).toBe(false)
+    expect(checkbox).not.toBeChecked()
+    expect(span).not.toHaveClass('Checkbox--indeterminate')
+    expect(checkbox).not.toHaveAttribute('aria-checked')
+    expect(dashPath).not.toBeInTheDocument()
+    expect(checkmarkPath).toBeInTheDocument()
+  })
+
+  it('has indeterminate state when indeterminate prop is true', () => {
+    const {getByRole, container} = render(<Checkbox value="test" indeterminate onChange={jest.fn()} />)
+
+    const checkbox = getByRole('checkbox') as HTMLInputElement
+
+    const span = container.querySelector('span.Checkbox')
+    const checkmarkPath = container.querySelector('path.Checkbox-checkmark-path')
```

Comment:
Do you think this might be too much implementation detail? Is it needed for coverage? Wondering if just checking the classname is sufficient?

Reply:
As discussed with @rezrah, we'll keep this as-is for now and add a follow-up task to add some way of uniquely identifying the dash icon to the `Checkbox` component, at which point we can refactor this test.